src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskAssignByIdControllerTest.java

```diff
@@ -71,5 +72,83 @@ public void should_return_a_204_when_assigning_a_task_by_id() {
+        Response result = restApiActions.post(
+            ENDPOINT_BEING_TESTED,
+            task.get("taskId"),
+            new AssigneeRequest(assigneeId),
+            APPLICATION_JSON_VALUE,
+            APPLICATION_JSON_VALUE,
+            authorizationHeadersProvider.getLawFirmAAuthorization()
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.UNAUTHORIZED.value())
+            .contentType(APPLICATION_JSON_VALUE)
+            .body("timestamp", lessThanOrEqualTo(LocalDateTime.now()
+                                                     .format(DateTimeFormatter.ofPattern(DATE_TIME_FORMAT))))
+            .body("error", equalTo(HttpStatus.UNAUTHORIZED.getReasonPhrase()))
+            .body("status", equalTo(HttpStatus.UNAUTHORIZED.value()))
+            .body("message", equalTo("User did not have sufficient permissions to perform this action"));
+    }
+
+    @Test
+    public void should_return_a_403_when_the_assigner_does_not_have_manage_permission() {
+        String noManagePermission = "Read,Refer,Own,Cancel";
+        Map<String, String> task = common.setupTaskAndRetrieveIds(noManagePermission);
+
+        String assigneeId = getAssigneeId(authorizationHeadersProvider.getCaseworkerAAuthorizationOnly());
+        String taskId = task.get("taskId");
+        Response result = restApiActions.post(
+            ENDPOINT_BEING_TESTED,
+            taskId,
+            new AssigneeRequest(assigneeId),
+            APPLICATION_JSON_VALUE,
+            APPLICATION_JSON_VALUE,
+            authorizationHeadersProvider.getTribunalCaseworkerAAuthorization()
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.FORBIDDEN.value())
+            .contentType(APPLICATION_JSON_VALUE)
+            .body("timestamp", lessThanOrEqualTo(LocalDateTime.now()
+                                                     .format(DateTimeFormatter.ofPattern(DATE_TIME_FORMAT))))
+            .body("error", equalTo(HttpStatus.FORBIDDEN.getReasonPhrase()))
+            .body("status", equalTo(HttpStatus.FORBIDDEN.value()))
+            .body("message",
+                  equalTo("User did not have sufficient permissions to assign task with id: " + taskId));
+    }
+
+    @Test
+    public void should_return_a_403_when_the_assignee_does_not_have_execute_or_own_permissions() {
+        String noOwnPermission = "Read,Refer,Manage,Cancel";
+        Map<String, String> task = common.setupTaskAndRetrieveIds(noOwnPermission);
+
+        String assigneeId = getAssigneeId(authorizationHeadersProvider.getCaseworkerAAuthorizationOnly());
+        String taskId = task.get("taskId");
+        Response result = restApiActions.post(
+            ENDPOINT_BEING_TESTED,
+            taskId,
+            new AssigneeRequest(assigneeId),
+            APPLICATION_JSON_VALUE,
+            APPLICATION_JSON_VALUE,
+            authorizationHeadersProvider.getTribunalCaseworkerAAuthorization()
+        );
+
```

Comment:
A user cannot assign a task to themselves. This will return 403 for execute or own permissions but the user be different for complete correctness of the journey. I.e.

authorizationHeadersProvider.getCaseworkerAAuthorizationOnly() = authorizationHeadersProvider.getTribunalCaseworkerAAuthorization()

Reply:
Not an issue as a user can assign a task to themselves